In [1]:
import pandas as pd
import numpy as np
columns = ["ID", "RI", "Na", "Mg", "Al", "Si", "K", "Ca", "Ba", "Fe", "Type"]
df = pd.read_csv("https://archive.ics.uci.edu/ml/machine-learning-databases/glass/glass.data", header=None, names=columns)
print(df)

      ID       RI     Na    Mg    Al     Si     K    Ca    Ba    Fe  Type
0      1  1.52101  13.64  4.49  1.10  71.78  0.06  8.75  0.00  0.00     1
1      2  1.51761  13.89  3.60  1.36  72.73  0.48  7.83  0.00  0.00     1
2      3  1.51618  13.53  3.55  1.54  72.99  0.39  7.78  0.00  0.00     1
3      4  1.51766  13.21  3.69  1.29  72.61  0.57  8.22  0.00  0.00     1
4      5  1.51742  13.27  3.62  1.24  73.08  0.55  8.07  0.00  0.00     1
5      6  1.51596  12.79  3.61  1.62  72.97  0.64  8.07  0.00  0.26     1
6      7  1.51743  13.30  3.60  1.14  73.09  0.58  8.17  0.00  0.00     1
7      8  1.51756  13.15  3.61  1.05  73.24  0.57  8.24  0.00  0.00     1
8      9  1.51918  14.04  3.58  1.37  72.08  0.56  8.30  0.00  0.00     1
9     10  1.51755  13.00  3.60  1.36  72.99  0.57  8.40  0.00  0.11     1
10    11  1.51571  12.72  3.46  1.56  73.20  0.67  8.09  0.00  0.24     1
11    12  1.51763  12.80  3.66  1.27  73.01  0.60  8.56  0.00  0.00     1
12    13  1.51589  12.88  3.43  1.40  

In [2]:
df.head()
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 214 entries, 0 to 213
Data columns (total 11 columns):
ID      214 non-null int64
RI      214 non-null float64
Na      214 non-null float64
Mg      214 non-null float64
Al      214 non-null float64
Si      214 non-null float64
K       214 non-null float64
Ca      214 non-null float64
Ba      214 non-null float64
Fe      214 non-null float64
Type    214 non-null int64
dtypes: float64(9), int64(2)
memory usage: 18.5 KB


ID          RI          Na          Mg          Al          Si  \
count  214.000000  214.000000  214.000000  214.000000  214.000000  214.000000   
mean   107.500000    1.518365   13.407850    2.684533    1.444907   72.650935   
std     61.920648    0.003037    0.816604    1.442408    0.499270    0.774546   
min      1.000000    1.511150   10.730000    0.000000    0.290000   69.810000   
25%     54.250000    1.516523   12.907500    2.115000    1.190000   72.280000   
50%    107.500000    1.517680   13.300000    3.480000    1.360000   72.790000   
75%    160.750000    1.519157   13.825000    3.600000    1.630000   73.087500   
max    214.000000    1.533930   17.380000    4.490000    3.500000   75.410000   

                K          Ca          Ba          Fe        Type  
count  214.000000  214.000000  214.000000  214.000000  214.000000  
mean     0.497056    8.956963    0.175047    0.057009    2.780374  
std      0.652192    1.423153    0.497219    0.097439    2.103739  
min      0.000000    5.430000    0.000000    0.000000    1.000000  
25%      0.122500    8.240000    0.000000    0.000000    1.000000  
50%      0.555000    8.600000    0.000000    0.000000    2.000000  
75%      0.610000    9.172500    0.000000    0.100000    3.000000  
max      6.210000   16.190000    3.150000    0.510000    7.000000

As per the current results, no data is missing in the given data set. 214 rows * 11 columns

In [3]:
df['Type'].value_counts()

2    76
1    70
7    29
3    17
5    13
6     9
Name: Type, dtype: int64

df['Type'] is a categorical attribute. df['Type'] in my understanding is the Y i.e. the Label, all other columns are the X i.e attributes. Let us divide the dataframe into X and Ys

In [4]:
df1 = df[["ID", "RI", "Na", "Mg", "Al", "Si", "K", "Ca", "Ba", "Fe"]]
df2 = df[["Type"]]

In [5]:
from sklearn.model_selection import train_test_split
#X_train, X_test, y_train, y_test = train_test_split(df1, df2, test_size=0.30, random_state=42)

In [44]:
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score
from sklearn.model_selection import KFold
import numpy
kf = KFold(n_splits=5)
kf.get_n_splits(df1)
print(kf)
a = pd.DataFrame()
b = []
b = np.array(b) # Initialize an empty numpy nd array without giving any size
clf = GaussianNB()
for train_index, test_index in kf.split(df1):
    #print("TRAIN:", train_index, "TEST:", test_index)
    X_train, X_test = df1.iloc[train_index], df1.iloc[test_index]
    y_train, y_test = df2.iloc[train_index], df2.iloc[test_index]
    clf.fit(X_train, y_train)
    y = clf.predict(X_test)
    a = a.append(y_test)
    b = numpy.append(b,y)
a = a.values # to convert a dataframe to numpy ndarray
print(accuracy_score(a,b))

KFold(n_splits=5, random_state=None, shuffle=False)
[[1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [3]
 [5]
 [5]
 [5]
 [5]
 [5]
 [5]
 [5]
 [5]
 [5]
 [5]
 [5]
 [5]
 [5]
 [6]
 [6]
 [6]
 [6]
 [6]
 [6]
 [6]
 [6]
 [6]
 [7]
 [7]
 [7]
 [7]
 [7

/anaconda3/lib/python3.7/site-packages/sklearn/utils/validation.py:578: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)
